src/tests/tests.luau

```diff
@@ -32,6 +32,36 @@ function tests.assertEquals<T>(x: T, y: T)
 	end
 end
 
+function tests.assertTablesEquivalent(x: table, y: table)
```

Comment:
`table` isn't actually a valid Luau type. There's some other iffy parts of this code but it's probably easier for me to just dump a version of this on you that I wrote for a game :)

```lua
local function shallowEqual<T>(x: T, y: T): boolean
	if x == y then
		return true
	end

	if typeof(x) ~= typeof(y) then
		return false
	end

	if typeof(x) == "table" then
		assert(typeof(y) == "table", "Luau")

		if #x ~= #y then
			return false
		end

		for key in x do
			if x[key] ~= y[key] then
				return false
			end
		end

		for key in y do
			if x[key] ~= y[key] then
				return false
			end
		end
	end

	return true
end
```

Reply:
Benefits:
- Enforces inputs are the exact same type, because if they're not then we've already failed
- No `pairs`